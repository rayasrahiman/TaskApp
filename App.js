import React, {useEffect} from 'react';
import {SafeAreaView, StyleSheet, BackHandler, LogBox} from 'react-native';
import {combineReducers, createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {Provider} from 'react-redux';

import myFirstReducer from './redux/reducer';
import rootSaga from './redux/sagas';
import MainNavigator from './navigation/NavigationScreen';
import {Colors} from './constants/colors';

const sagaMiddleware = createSagaMiddleware();
const rootReducer = combineReducers({myFirstReducer});
const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(rootSaga);

const App = () => {
  useEffect(() => {
    LogBox.ignoreLogs(['new NativeEventEmitter']);
    LogBox.ignoreAllLogs();
    // const backHandler = BackHandler.addEventListener(
    //   'hardwareBackPress',
    //   () => true,
    // );
    // return () => backHandler.remove();
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <Provider store={store}>
        <MainNavigator />
      </Provider>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.white,
  },
});

export default App;
