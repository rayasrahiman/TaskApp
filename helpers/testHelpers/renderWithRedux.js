import React from 'react';
import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import {createStore, combineReducers, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';

import myFirstReducer from '../../redux/reducer';
import rootSaga from '../../redux/sagas';

export const renderWithRedux = renderedComp => {
  const sagaMiddleware = createSagaMiddleware();
  const rootReducer = combineReducers({myFirstReducer});

  const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
  sagaMiddleware.run(rootSaga);

  return render(<Provider store={store}>{renderedComp}</Provider>);
};
