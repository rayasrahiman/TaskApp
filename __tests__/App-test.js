import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';

import App from '../App';
import {renderWithRedux} from '../helpers/testHelpers/renderWithRedux';

beforeEach(jest.spyOn(console, 'error').mockImplementation(() => undefined));
afterEach(() => jest.clearAllMocks());

describe('App', () => {
  test('should render correctly', () => {
    const wrapper = renderWithRedux(<App />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
