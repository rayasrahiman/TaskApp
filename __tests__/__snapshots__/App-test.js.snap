// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          Object {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={true}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="Pager"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "flex": 1,
              }
            }
          >
            <View>
              <RCTScrollView
                data={
                  Array [
                    Object {
                      "id": 1,
                      "img": Object {
                        "testUri": "../../../assets/img1.jpg",
                      },
                      "subtitle": "Stay smart connected and protected.",
                      "title": "Take Control of your home's water.",
                    },
                    Object {
                      "id": 2,
                      "img": Object {
                        "testUri": "../../../assets/img2.jpg",
                      },
                      "subtitle": "Get status information, home or away 24/7.",
                      "title": "Manage systems on the go.",
                    },
                    Object {
                      "id": 3,
                      "img": Object {
                        "testUri": "../../../assets/img3.jpg",
                      },
                      "subtitle": "Receive alerts sent to your phone and if you choose your trusted pros.",
                      "title": "We'll let you know what to do, when.",
                    },
                    Object {
                      "id": 4,
                      "img": Object {
                        "testUri": "../../../assets/img4.jpg",
                      },
                      "subtitle": "Peace of mins at your fingretips.",
                      "title": "Worry a lot less, enjoy a lot more.",
                    },
                  ]
                }
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={true}
                keyExtractor={[Function]}
                onContentSizeChange={[Function]}
                onLayout={[Function]}
                onMomentumScrollBegin={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                pagingEnabled={true}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                showsHorizontalScrollIndicator={false}
                stickyHeaderIndices={Array []}
                testID="viewPager"
                viewabilityConfigCallbackPairs={Array []}
              >
                <View>
                  <View
                    onLayout={[Function]}
                    style={
                      Array [
                        Object {
                          "flexDirection": "row",
                        },
                        null,
                      ]
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "height": 667,
                            "width": 750,
                          }
                        }
                      >
                        <Image
                          resizeMode="cover"
                          source={
                            Object {
                              "testUri": "../../../assets/img1.jpg",
                            }
                          }
                          style={
                            Object {
                              "height": 667,
                              "width": 750,
                            }
                          }
                        />
                      </View>
                      <View>
                        <Image
                          resizeMode="contain"
                          source={
                            Object {
                              "testUri": "../../../assets/Logo.png",
                            }
                          }
                          style={
                            Object {
                              "height": 200,
                              "width": 112.5,
                            }
                          }
                        />
                      </View>
                      <View>
                        <View
                          style={
                            Object {
                              "marginHorizontal": 0,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#244379",
                                "fontSize": 46.5,
                                "fontWeight": "bold",
                                "marginHorizontal": 13.5,
                              }
                            }
                          >
                            Take Control of your home's water.
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "marginHorizontal": 0,
                              "width": 750,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#838383",
                                "fontSize": 40,
                                "fontWeight": "bold",
                                "marginHorizontal": 13.5,
                                "marginVertical": 66.5,
                              }
                            }
                          >
                            Stay smart connected and protected.
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={
                      Array [
                        Object {
                          "flexDirection": "row",
                        },
                        null,
                      ]
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "height": 667,
                            "width": 750,
                          }
                        }
                      >
                        <Image
                          resizeMode="cover"
                          source={
                            Object {
                              "testUri": "../../../assets/img2.jpg",
                            }
                          }
                          style={
                            Object {
                              "height": 667,
                              "width": 750,
                            }
                          }
                        />
                      </View>
                      <View>
                        <Image
                          resizeMode="contain"
                          source={
                            Object {
                              "testUri": "../../../assets/Logo.png",
                            }
                          }
                          style={
                            Object {
                              "height": 200,
                              "width": 112.5,
                            }
                          }
                        />
                      </View>
                      <View>
                        <View
                          style={
                            Object {
                              "marginHorizontal": 0,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#244379",
                                "fontSize": 46.5,
                                "fontWeight": "bold",
                                "marginHorizontal": 13.5,
                              }
                            }
                          >
                            Manage systems on the go.
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "marginHorizontal": 0,
                              "width": 750,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#838383",
                                "fontSize": 40,
                                "fontWeight": "bold",
                                "marginHorizontal": 13.5,
                                "marginVertical": 66.5,
                              }
                            }
                          >
                            Get status information, home or away 24/7.
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={
                      Array [
                        Object {
                          "flexDirection": "row",
                        },
                        null,
                      ]
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "height": 667,
                            "width": 750,
                          }
                        }
                      >
                        <Image
                          resizeMode="cover"
                          source={
                            Object {
                              "testUri": "../../../assets/img3.jpg",
                            }
                          }
                          style={
                            Object {
                              "height": 667,
                              "width": 750,
                            }
                          }
                        />
                      </View>
                      <View>
                        <Image
                          resizeMode="contain"
                          source={
                            Object {
                              "testUri": "../../../assets/Logo.png",
                            }
                          }
                          style={
                            Object {
                              "height": 200,
                              "width": 112.5,
                            }
                          }
                        />
                      </View>
                      <View>
                        <View
                          style={
                            Object {
                              "marginHorizontal": 0,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#244379",
                                "fontSize": 46.5,
                                "fontWeight": "bold",
                                "marginHorizontal": 13.5,
                              }
                            }
                          >
                            We'll let you know what to do, when.
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "marginHorizontal": 0,
                              "width": 750,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#838383",
                                "fontSize": 40,
                                "fontWeight": "bold",
                                "marginHorizontal": 13.5,
                                "marginVertical": 66.5,
                              }
                            }
                          >
                            Receive alerts sent to your phone and if you choose your trusted pros.
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={
                      Array [
                        Object {
                          "flexDirection": "row",
                        },
                        null,
                      ]
                    }
                  >
                    <View>
                      <View
                        style={
                          Object {
                            "height": 667,
                            "width": 750,
                          }
                        }
                      >
                        <Image
                          resizeMode="cover"
                          source={
                            Object {
                              "testUri": "../../../assets/img4.jpg",
                            }
                          }
                          style={
                            Object {
                              "height": 667,
                              "width": 750,
                            }
                          }
                        />
                      </View>
                      <View>
                        <Image
                          resizeMode="contain"
                          source={
                            Object {
                              "testUri": "../../../assets/Logo.png",
                            }
                          }
                          style={
                            Object {
                              "height": 200,
                              "width": 112.5,
                            }
                          }
                        />
                      </View>
                      <View>
                        <View
                          style={
                            Object {
                              "marginHorizontal": 0,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#244379",
                                "fontSize": 46.5,
                                "fontWeight": "bold",
                                "marginHorizontal": 13.5,
                              }
                            }
                          >
                            Worry a lot less, enjoy a lot more.
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "marginHorizontal": 0,
                              "width": 750,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "color": "#838383",
                                "fontSize": 40,
                                "fontWeight": "bold",
                                "marginHorizontal": 13.5,
                                "marginVertical": 66.5,
                              }
                            }
                          >
                            Peace of mins at your fingretips.
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "position": "relative",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "marginBottom": 26.5,
                      "marginHorizontal": 7.5,
                    },
                    Object {
                      "flexDirection": "row",
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "#244379",
                      "borderColor": "#244379",
                      "borderRadius": 16.75,
                      "borderWidth": 5,
                      "height": 33.5,
                      "margin": 3,
                      "width": 33.5,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "backgroundColor": "#838383",
                      "borderColor": "#838383",
                      "borderRadius": 16.75,
                      "borderWidth": 5,
                      "height": 33.5,
                      "margin": 3,
                      "width": 33.5,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "backgroundColor": "#838383",
                      "borderColor": "#838383",
                      "borderRadius": 16.75,
                      "borderWidth": 5,
                      "height": 33.5,
                      "margin": 3,
                      "width": 33.5,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "backgroundColor": "#838383",
                      "borderColor": "#838383",
                      "borderRadius": 16.75,
                      "borderWidth": 5,
                      "height": 33.5,
                      "margin": 3,
                      "width": 33.5,
                    }
                  }
                />
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "marginBottom": 26.5,
                    "marginHorizontal": 7.5,
                    "opacity": 1,
                  }
                }
                testID="btnRight"
              >
                <Text
                  style={
                    Object {
                      "color": "#838383",
                      "fontSize": 40,
                    }
                  }
                >
                  Skip
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </RNCSafeAreaProvider>
</RCTSafeAreaView>
`;
